controllers/menuController.js

```diff
@@ -0,0 +1,96 @@
+const Menu = require("./../models/Menu");
+const jwt = require("jsonwebtoken");
+
+const check = await User.findOne({ email }).lean().exec();
+
+if (check) {
+    return res.status(400).json({
+        message:
+            "The email address already exists, try with a different email address.",
+    });
+}
+
+
+// Get all menu item
+
+const getAllMenu = async (req, res) => {
+    try {
+        const allMenuItems = await Menu.find();
+
+        if (!allMenuItems?.length) {
+            return res.status(400).json({ message: "No menu available" });
+        }
+
+        res.status(200).json({ success: true, data: allMenuItems });
+
+    } catch (error) {
+        console.error('Error getting all menu items:', error);
+        res.status(500).json({ success: false, error: 'Internal server error' });
+    }
+};
+
+
+
+// add new menu item
+
+const addMenuItem = async (req, res) => {
+    try {
+
+        const {
+            itemImage,
+            itemName,
+            originalPrice,
+            itemDescription,
+            merchantId,
+            cuisineTypeId,
```

Comment:
Only logged in merchant can create a menu. I have already attached the merchantId automatically in jwt. So, to get that, the merchant has to be logged in and you will be able to get the merchantId using` req.merchantId`